include/CMakeLists.txt

```diff
@@ -0,0 +1,22 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
The `include` folder is for public header files that will be exported and installed. I'm also fine to move them next to the cc files. In that case, I think it would be better to add `_internal.h` suffix to non-exported header files.

Reply:
I gave this a second thought: I think in the Java implementation there is a purpose of having an api/ module: Whatever is there isn't expected to change out of the Blue. Once you have anything in the spec you have to deprecate first in one of the upcoming releases and you can drop only in the following major release.

I think we can follow this as well: Have an api/ folder for the headers that we will consider the part of the lib the users should interact with. In Java these are interfaces if I'm not mistaken and then the actual implementations are part of the other modules like core/ etc. For instance Table is in the api/ and is a pure interface while inherent classes like BaseTable are in core/
I find this something we should also follow.